### PR TITLE
bpo-40192: Use thread_cputime for time.thread_time to improve resolution

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue40192

### CODE REVIEW

File: Modules/timemodule.c [hunk 1]
```diff
@@ -1343,6 +1347,26 @@ _PyTime_GetThreadTimeWithInfo(_PyTime_t *tp, _Py_clock_info_t *info)
     *tp = t;
     return 0;
 }
```
Comment (vstinner) [inline Modules/timemodule.c:1349]: Please add a newline for readability.

File: Modules/timemodule.c [hunk 2]
```diff
@@ -1343,6 +1347,26 @@ _PyTime_GetThreadTimeWithInfo(_PyTime_t *tp, _Py_clock_info_t *info)
...
+#elif defined(_AIX)
+#define HAVE_THREAD_TIME
+static int
+_PyTime_GetThreadTimeWithInfo(_PyTime_t *tp, _Py_clock_info_t *info)
+{
+    thread_cputime_t tc;
```
Comment (vstinner) [inline Modules/timemodule.c:1359]: Please add a comment explaining why thread_cputime() is preferred over standard clock_gettime(CLOCK_THREAD_CPUTIME_ID). Add "bpo-40192: " prefix, to add a reference to the issue.

File: Modules/timemodule.c [hunk 3]
```diff
@@ -1343,6 +1347,26 @@ _PyTime_GetThreadTimeWithInfo(_PyTime_t *tp, _Py_clock_info_t *info)
...
+        return -1;
+    }
+
+    if (info) {
+        info->implementation = "thread_cputime()";
+        info->monotonic = 1;
+        info->adjustable = 0;
+        info->resolution = 1e-9;
+    }
+    *tp = _PyTime_FromNanoseconds(tc.stime);
```
Comment (vstinner) [inline Modules/timemodule.c]: That's wrong: you must return stime+utime, https://bugs.python.org/issue40192#msg368947

**Review by vstinner — CHANGES_REQUESTED** [review]

File: Modules/timemodule.c [hunk 4]
```diff
@@ -1344,6 +1348,30 @@ _PyTime_GetThreadTimeWithInfo(_PyTime_t *tp, _Py_clock_info_t *info)
...
 }
 
+#elif defined(_AIX)
+#define HAVE_THREAD_TIME
+static int
+_PyTime_GetThreadTimeWithInfo(_PyTime_t *tp, _Py_clock_info_t *info)
+{
+    /* bpo-40192: On AIX, thread_cputime is preffered because
+    it is using nano-second time resolution (compared to 10ms)
+    resolution of clock_gettime(CLOCK_THREAD_CPUTIME_ID) */
```
Comment (vstinner) [inline Modules/timemodule.c]: typo: preferred

```suggestion
    /* bpo-40192: On AIX, thread_cputime() is preferred: it has nanosecond
       resolution, whereas clock_gettime(CLOCK_THREAD_CPUTIME_ID)
       has a resolution of 10 ms. */
```

File: Doc/whatsnew/3.9.rst [hunk 5]
```diff
@@ -371,6 +371,12 @@ finalization crashed with a Python fatal error if a daemon thread was still
 running.
 (Contributed by Victor Stinner in :issue:`37266`.)
 
+time
+----
+
+:func:`~time.thread_time` now output in nanosecond resolution via
+``thread_cputime`` on AIX. (Contributed by Batuhan Taskaya in :issue:`40192`)
```
Comment (vstinner) [inline Doc/whatsnew/3.9.rst]: Oh, I proposed a different text, but it seems like my comment is lost :-( I propose:

```
On AIX, :func:`~time.thread_time` is now implemented with ``thread_cputime()``
which has nanosecond resolution, than than
``clock_gettime(CLOCK_THREAD_CPUTIME_ID)`` which has a resolution of 10 ms.
(Contributed by Batuhan Taskaya in :issue:`40192`)
```

Use the same text in the NEWS entry, but replace "(Contributed by Batuhan Taskaya in :issue:`40192`)" with "Patch by Batuhan Taskaya.".

**Comment by isidentical** [comment]
I have made the requested changes; please review again

**Comment by isidentical** [comment]
I have made the requested changes; please review again

**Comment by vstinner** [comment]
Merged, thanks @isidentical!

**Comment by isidentical** [comment]
Thank you for the review @vstinner!

**Pull request merged**